app/code/Magento/Catalog/Block/Product/ProductList/Promotion.php

```diff
@@ -38,6 +40,7 @@ public function __construct(
         \Magento\Catalog\Model\Layer\Resolver $layerResolver,
         CategoryRepositoryInterface $categoryRepository,
         \Magento\Framework\Url\Helper\Data $urlHelper,
+        CollectionLoader $collectionLoader,
```

Comment:
New injections should come last in the construct and should also default to null so as to not break backwards compatibility. More information and examples can be found at http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/index.html#adding-a-constructor-parameter